source/DistanceMap.cpp

```diff
@@ -80,39 +80,38 @@ DistanceMap::DistanceMap(const PlayerInfo &player, const System *center)
 // pathfinding will stop once a path to the destination is found.
 // If a player is given, the path will only include systems that the
 // player has visited.
-DistanceMap::DistanceMap(const Ship &ship, const System *destination, const PlayerInfo *player)
-	: player(player), source(ship.GetSystem()), center(destination)
+DistanceMap::DistanceMap(const System &center, const System &destination, const PlayerInfo *player)
+	: player(player), center(&center), destination(&destination)
 {
-	if(!source || !destination)
-		return;
-
-	Init(&ship);
+	if(player)
+		Init(player->Flagship());
+	else
+		Init();
 }
 
 
 
-// Find out if the given system is reachable.
-bool DistanceMap::HasRoute(const System *system) const
+DistanceMap::DistanceMap(const Ship &ship, const System &destination, const PlayerInfo *player)
+	: player(player), center(ship.GetSystem()), destination(&destination)
 {
-	return route.contains(system);
+	Init(&ship);
 }
 
 
 
-// Find out how many days away the given system is.
-int DistanceMap::Days(const System *system) const
+// Find out if the given system is reachable
+bool DistanceMap::HasRoute(const System &target) const
 {
-	auto it = route.find(system);
-	return (it == route.end() ? -1 : it->second.days);
+	return route.contains(&target);
 }
 
 
 
-// Starting in the given system, what is the next system along the route?
-const System *DistanceMap::Route(const System *system) const
+// Find out how many days away the given system is.
+int DistanceMap::Days(const System &target) const
 {
-	auto it = route.find(system);
-	return (it == route.end() ? nullptr : it->second.next);
+	auto it = route.find(&target);
+	return (it == route.end() ? -1 : it->second.days);
```

Comment:
Maybe use an std::optional instead of the -1 for no value?

Reply:
(I thought this comment got through?)

This would be a change independent of this branch